sources/core/Stride.Core/Diagnostics/ProfilingMessage.cs

```diff
@@ -56,7 +57,25 @@ public ProfilingMessage(int profileId, [NotNull] ProfilingKey profilingKey, Prof
         /// </summary>
         /// <value>The type of the profile.</value>
         public ProfilingMessageType ProfilingType { get; }
- 
+
+        /// <summary>
+        /// Event message.
+        /// </summary>
+        public ProfilingEventMessage? Message { get; }
+
+        /// <summary>
+        /// Text of the log.
+        /// </summary>
+        public override string Text
+        {
+            get => Message?.ToString();
+            set
+            {
+                // ignore null set in base constructor call, otherwise throw
+                if (value != null) throw new NotSupportedException("Set Message instead.");
+            }
+        }
```

Comment:
This isn't great, it could create a new message with the content of the string provided instead of throwing ? I know you would prefer users to set the message instead but it does make the api more fragile/obtuse.

Reply:
I didn't think of just setting the text message - sounds good